Cython/Compiler/Visitor.py

```diff
@@ -370,9 +370,18 @@ def exit_scope(self):
         self.env_stack.pop()
 
     def visit_FuncDefNode(self, node):
+        # note that "return_type_annotation" is only on subclasses
+        # FIXME just "return_type_annotation" or all outer_attrs?
+        attrs = [attr for attr in node.child_attrs if attr != "return_type_annotation"]
         self.enter_scope(node, node.local_scope)
-        self._process_children(node)
+        self._process_children(node, attrs)
+        outer_scope = self.current_env().outer_scope
         self.exit_scope()
+
+        if getattr(node, "return_type_annotation", None) is not None:
```

Comment:
I wouldn't mind if all `FuncDefNode`s had a `return_type_annotation`, defaulting to `None`.